### PR TITLE
JENKINS-65393: Help for CleanWhenUnstable

### PR DESCRIPTION
**Pull request by jebitok-dev**
- I added help to the Workspace Cleanup for the CleanWhenUnstable pipeline step. 
- Attached is my [issue](https://issues.jenkins.io/browse/JENKINS-65393) on Jira
- Please  @MarkEWaite, @Stackscribe, or @kwhetstone, I have created a new pull request I'm requesting for your code review.
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fix the issue

### CODE REVIEW

**Comment by MarkEWaite** [comment]
See [JENKINS-65393](https://issues.jenkins.io/browse/JENKINS-65393) for the issue

**Comment by MarkEWaite** [comment]
@jebitok-dev and I had a sharing session that extended this pull request with additional changes to provide more help from the snippet generator.  We were unable to find a way to add help to the Pipeline steps reference.  That help may come later.

She will evaluate the draft we created at https://github.com/MarkEWaite/ws-cleanup-plugin/commits/develop and if those changes are useful, she'll consider including them in this pull request.

**Comment by kwhetstone** [comment]
@jebitok-dev how's the reviewing going for this documentation?

**Comment by jebitok-dev** [comment]
> @jebitok-dev how's the reviewing going for this documentation?
I have made the changes that we had discussed but I'm having issues/errors running ``$  mvn -Djenkins.version=2.277.2 hpi:run`` locally to compile plugin & check if the help shows up on Jenkins pipeline syntax. I was wondering if I can commit the code then if possible someone can confirm from their end before either merging or closing the Pull Request.

**Comment by jebitok-dev** [comment]
@kwhetstone I have confirmed that the help is showing on the pipeline steps. We had difficultly identifying how it can be made to show on the pipeline step-reference for this plugin.

File: src/main/webapp/help/cleanWhenStatusIs.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<div>
+    Remove workspace contents based on the result of the build.
+    Allows the Pipeline to not clean the workspace contents if the build was unstable or failed and needs the contents of the workspace to investigate the failure.
+    
+    <p>
+    Choices include:
+    <ul>
+    <li><strong>Success</strong>: If enabled, the workspace will be cleaned on successful builds.  Enabled by default.</li>
```
Comment (StackScribe) [inline src/main/webapp/help/cleanWhenStatusIs.html:8]:
```suggestion
    <li><strong>Success</strong>: If enabled, the workspace is cleaned on successful builds.  Enabled by default.</li>
```

File: src/main/webapp/help/cleanWhenStatusIs.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    <li><strong>Unstable</strong>: If enabled, the workspace will be cleaned on unstable builds.  Enabled by default.</li>
```
Comment (StackScribe) [inline src/main/webapp/help/cleanWhenStatusIs.html:9]:
```suggestion
    <li><strong>Unstable</strong>: If enabled, the workspace is cleaned on unstable builds.  Enabled by default.</li>
```

File: src/main/webapp/help/cleanWhenStatusIs.html [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    <li><strong>Failure</strong>: If enabled, the workspace will be cleaned on failed builds.  Enabled by default.</li>
```
Comment (StackScribe) [inline src/main/webapp/help/cleanWhenStatusIs.html:10]:
```suggestion
    <li><strong>Failure</strong>: If enabled, the workspace is cleaned on failed builds.  Enabled by default.</li>
```

File: src/main/webapp/help/cleanWhenStatusIs.html [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    <li><strong>Not built</strong>: If enabled, the workspace will be cleaned on not built builds.  <strong>Not Built needs more clarification!</strong>  Enabled by default.</li>
```
Comment (StackScribe) [inline src/main/webapp/help/cleanWhenStatusIs.html:11]:
```suggestion
    <li><strong>Not built</strong>: If enabled, the workspace is cleaned on not built builds.  <strong>Not Built needs more clarification!</strong>  Enabled by default.</li>
```

File: src/main/webapp/help/cleanWhenStatusIs.html [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    <li><strong>Aborted</strong>: If enabled, the workspace will be cleaned on aborted builds.  <strong>When Aborted needs more clarification</strong>  Enabled by default.</li>
```
Comment (StackScribe) [inline src/main/webapp/help/cleanWhenStatusIs.html:12]:
```suggestion
    <li><strong>Aborted</strong>: If enabled, the workspace is cleaned on aborted builds.  <strong>When Aborted needs more clarification</strong>  Enabled by default.</li>
```

**Review by StackScribe — COMMENTED** [review]
Nicely done!  I made some word-smithing suggestions.  Using the present tense is a little cleaner than the future tense.

**Review by MarkEWaite — APPROVED** [review]

**Review by basil — APPROVED** [review]
Thanks for the PR!

**Pull request merged**